store-implementation/accumulo-store/src/test/java/uk/gov/gchq/gaffer/accumulostore/MiniAccumuloStore.java

```diff
@@ -0,0 +1,176 @@
+ * existing user will be overwritten.
+ */
+public class MiniAccumuloStore extends AccumuloStore {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MiniAccumuloStore.class);
+
+    private static final HashMap<String, MiniAccumuloCluster> CLUSTER_INSTANCES = new HashMap<>();
+    private static final HashMap<String, File> ACCUMULO_DIRECTORIES = new HashMap<>();
+
+    private static final String ROOT_PASSWORD_DEFAULT = "password";
+    private static final int DEFAULT_ZOOKEEPER_PORT = 2181;
+
+    private static final String ROOT_USER = "root";
+    private static final String VISIBILITIES_PROPERTY = "accumulo.mini.visibilities";
+    private static final String ROOT_PASSWORD_PROPERTY = "accumulo.mini.root.password";
+    private static final String ACCUMULO_DIRECTORY_PROPERTY = "accumulo.mini.directory";
+
+
+    @Override
+    public void preInitialise(final String graphId, final Schema schema, final StoreProperties properties) throws StoreException {
+        super.preInitialise(graphId, schema, properties);
+
+        synchronized (this) {
+            if (getCluster() == null) {
+                try {
+                    createCluster();
+                } catch (final InterruptedException | IOException e) {
+                    throw new StoreException("Failed to start accumulo cluster", e);
+                }
+            }
+        }
+
+
+        try {
+            ensureUserExists(getCluster());
+        } catch (final AccumuloException | AccumuloSecurityException e) {
+            throw new StoreException("Failed to ensure user was added", e);
+        }
+
+    }
+
+    private MiniAccumuloCluster getCluster() {
+        return CLUSTER_INSTANCES.get(getProperties().getInstance());
+    }
+
+    private File getAccumuloDirectory() {
+        return ACCUMULO_DIRECTORIES.get(getProperties().getInstance());
+    }
+
+    private void ensureUserExists(final MiniAccumuloCluster mac) throws AccumuloSecurityException, AccumuloException {
+        String userName = getProperties().getUser();
+        String rootPassword = getProperties().get(ROOT_PASSWORD_PROPERTY, ROOT_PASSWORD_DEFAULT);
+
+        // Ensure user exists
+        synchronized (this) {
+            Set<String> currentUsers = mac.getConnector(ROOT_USER, rootPassword).securityOperations().listLocalUsers();
+            if (!currentUsers.contains(userName)) {
+                mac.getConnector(ROOT_USER, rootPassword).securityOperations().createLocalUser(
+                        userName, new PasswordToken(getProperties().getPassword())
+                );
+                mac.getConnector(ROOT_USER, rootPassword).securityOperations()
+                        .grantSystemPermission(userName, SystemPermission.CREATE_NAMESPACE);
+                mac.getConnector(ROOT_USER, rootPassword).securityOperations()
+                        .grantSystemPermission(userName, SystemPermission.CREATE_TABLE);
+            }
+        }
+
+
+        // Add Auths
```

Comment:
Consider extracting to private method over commenting

Reply:
Will do

Reply:
Actually when I tried it, it looked messy given I had to pass the MiniAccumulo cluster, root password and username to a short private method.